generation/core/generator.js

```diff
@@ -81,13 +89,19 @@ module.exports = function({
 	}
 
 	function createMethod(classJson, json) {
+		const args = json.args.map(({ name }) => t.identifier(name));
+
+		if (!json.static) {
+			args.unshift(t.identifier("element"));
+		}
+
 		const m = t.classMethod(
 			"method",
 			t.identifier(methodNameToSnakeCase(json.name)),
-			json.args.map(({ name }) => t.identifier(name)),
+			args,
 			t.blockStatement(createMethodBody(classJson, json)),
 			false,
-			json.static
+			true
```

Comment:
will this mean every method is static now ?

Reply:
Then why passing this boolean anyway ? why not get rid of it ?

Reply:
https://martinfowler.com/bliki/FlagArgument.html

Reply:
This is a method of `babel-types`, I could write a wrapper like `createStaticClassMethod`, would that be more readable for you?

Every method is static, I choose to deal with instance methods a little different, I just let the target be passed in as the first argument. My point for this is that we have the class-like behavior more on the caller-side of this function and I don't want to duplicate it in any way. Also, the calls are more similar now than they would have been otherwise. If we want to change this somewhen I would propose we rethink this at a point in time when we get rid of these wrapper classes and access the generated code "directly", like in this proposal: #563